src/error.rs

```diff
@@ -54,23 +54,27 @@ impl std::error::Error for Error {
 }
 
 #[cfg(feature = "instrumentation")]
-pub static IO_COUNTER_BEFORE_ERROR: AtomicUsize = AtomicUsize::new(usize::MAX);
+thread_local! {
```

Comment:
There was only one `instrumented` test so far. I've added another and they started interfering. Each test should have its own counter. Using TLS is fine because tests disable all background threads and process everything in the the test thread.